client/txn_test.go

```diff
@@ -288,7 +288,7 @@ func TestEndWriteRestartReadOnlyTransaction(t *testing.T) {
 			t.Errorf("expected error: %t, got error: %v", !success, err)
 		}
 		if !reflect.DeepEqual(expCalls, calls) {
-			t.Fatalf("expected %v, got %v", expCalls, calls)
+			t.Errorf("expected %v, got %v", expCalls, calls)
```

Comment:
I agree with @petermattis though I've found `Fatalf` annoying in some situations in which it prevents clean-ish shutdown of the test (often, you find yourself timing out because some task never gets drained, though these can be remedied by making the defers panic-safe).